src/Polly.Contrib.Simmy/AsyncMonkeyEngine.cs

```diff
@@ -7,14 +7,29 @@ namespace Polly.Contrib.Simmy
 {
     internal static class AsyncMonkeyEngine
     {
-        private static async Task<bool> ShouldInjectAsync(Context context, Func<Context, Task<double>> injectionRate, Func<Context, Task<bool>> enabled, bool continueOnCapturedContext)
+        private static async Task<bool> ShouldInjectAsync(
+            Context context, 
+            CancellationToken cancellationToken, 
+            Func<Context, CancellationToken, Task<double>> injectionRate, 
+            Func<Context, CancellationToken, Task<bool>> enabled, 
+            bool continueOnCapturedContext)
         {
-            if (!await enabled(context).ConfigureAwait(continueOnCapturedContext))
+            // to prevent execute config delegates if user cancels the whole execution before to start.
+            cancellationToken.ThrowIfCancellationRequested();
+
+            if (!await enabled(context, cancellationToken).ConfigureAwait(continueOnCapturedContext))
             {
                 return false;
             }
 
-            double injectionThreshold = await injectionRate(context).ConfigureAwait(continueOnCapturedContext);
+            // to prevent execute injectionRate config delegate if user cancels execution on enable configuration delegate.
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Agree, I think that's gonna cover cancellation for timing out just in case the user wrap chaos policies with a timeout policy, which would be highly likely.